### DISCUSSION TITLE
Creating a Model from a Schem

### DISCUSSION THREAD

**Discussion by iyumeg** [pydantic/pydantic#5093] [Question] [answered]
Pidantic makes it possible to pack a model into a schema using the schema (or schema_json) method, but the documentation does not mention a method that allows you to do the opposite, namely, convert the schema to a model and validate the data.

I form data models based on my business logic and let users enter data after which I want to validate the schema and save the data to the database.

Please tell me how can this be done. I know the library "datamodel-code-generator" which allows you to generate files with models based on a schema or swagger

**Comment by iyumeg** [comment]
it's kinda stupid to do smth like this

```python3
def create_model_from_schema(schema: dict) -> type:
    with TemporaryDirectory() as temporary_directory_name:
        temporary_directory = Path(temporary_directory_name)
        output = Path(temporary_directory / 'model.py')
        generate(
            json.dumps(schema),
            input_file_type=InputFileType.JsonSchema,
            input_filename="example.json",
            output=output,
        )
        form_module_spec = spec_from_file_location("model", output.absolute())
        form_module = module_from_spec(form_module_spec)
        form_module_spec.loader.exec_module(form_module)
        return getattr(form_module, "FormSchema")
```

**Reply by koxudaxi** [reply]
@iyumeg 
The project was created for code generation.
If you have a good idea for the use case, we can discuss the topic in [Issue](https://github.com/koxudaxi/datamodel-code-generator/issues) or [Discussions](https://github.com/koxudaxi/datamodel-code-generator/discussions) in `datamodel-code-generator` project.

**Comment by samuelcolvin** [comment, marked as answer]
`datamodel-code-generator` is the only solution I know of.
